src/rest_vol_dataset.c

```diff
@@ -5217,6 +5369,370 @@ RV_get_cmpd_subset_nmembers(hid_t src_type_id, hid_t dst_type_id, size_t *num_cm
+    const char       *elem_fmt_string     = NULL;
+    char             *out_string          = NULL;
+    char             *out_string_curr_pos = NULL;
+    ptrdiff_t         out_diff            = 0;
+    int               bytes_printed       = 0;
+    size_t            tgt_out_size        = 0;
+    size_t            parent_dtype_size;
+
+    if (!out)
+        FUNC_GOTO_ERROR(H5E_DATASET, H5E_BADVALUE, H5_ITER_ERROR,
+                        "invalid NULL pointer provided for output of vlen to JSON conversion");
+
+    if (!_in)
+        FUNC_GOTO_ERROR(H5E_DATASET, H5E_BADVALUE, H5_ITER_ERROR,
+                        "invalid NULL pointer provided for input of vlen to JSON conversion");
+
+    in = (const hvl_t *)_in;
+
+    if ((parent_type = H5Tget_super(dtype_id)) < 0)
+        FUNC_GOTO_ERROR(H5E_DATASET, H5E_BADVALUE, FAIL, "can't get base type of vlen type");
+
+    if ((parent_class = H5Tget_class(parent_type)) < 0)
+        FUNC_GOTO_ERROR(H5E_DATASET, H5E_CANTGET, FAIL, "can't get class of parent of vlen dtype");
+
+    if ((parent_dtype_size = H5Tget_size(parent_type)) == 0)
+        FUNC_GOTO_ERROR(H5E_DATASET, H5E_CANTGET, FAIL, "can't get size of vlen seq base type");
+
+    if ((out_string = (char *)calloc(DATASET_VLEN_JSON_BODY_DEFAULT_SIZE, 1)) == NULL)
+        FUNC_GOTO_ERROR(H5E_DATASET, H5E_CANTALLOC, FAIL, "can't allocate space for vlen data JSON");
+
+    *out_size           = DATASET_CREATION_PROPERTIES_BODY_DEFAULT_SIZE;
+    out_string_curr_pos = out_string;
+
+    /* Delineate start of list of sequences */
+    out_diff     = out_string_curr_pos - out_string;
+    tgt_out_size = (size_t)out_diff;
+    CHECKED_REALLOC(out_string, *out_size, tgt_out_size, out_string_curr_pos, H5E_DATASET, FAIL);
+    out_string = strcat(out_string, "[");
+    out_string_curr_pos += 1;
+
+    /* The amount of space needed to serialize each element can't be known ahead of time, so serialize each
+     * element into a temporary fixed-size buffer before copying it to the output buffer . */
+    for (size_t i = 0; i < nelems; i++) {
+        hvl_t seq = in[i];
+
+        /* Delineate start of this sequence */
+        out_diff     = out_string_curr_pos - out_string;
+        tgt_out_size = (size_t)(out_diff) + 1;
+        CHECKED_REALLOC(out_string, *out_size, tgt_out_size, out_string_curr_pos, H5E_DATASET, FAIL);
+        out_string = strcat(out_string, "[");
+        out_string_curr_pos += 1;
+
+        /* Stringify each element in a second fixed-size buffer, then copy to sequence buffer */
+        for (size_t j = 0; j < seq.len; j++) {
+#define MAX_SEQ_ELEM_BUF_SIZE 128
+            char  seq_elem_buffer[MAX_SEQ_ELEM_BUF_SIZE];
+            char *seq_elem_buffer_ptr = seq_elem_buffer;
+
+            memset(seq_elem_buffer, 0, MAX_SEQ_ELEM_BUF_SIZE);
+
+            switch (parent_class) {
+                case (H5T_INTEGER):
+                    if ((bytes_printed = snprintf(seq_elem_buffer_ptr, MAX_SEQ_ELEM_BUF_SIZE, "%d",
```

Comment:
This should at least verify that the parent datatype matches `H5T_NATIVE_INT` before trying to cast the buffer to `int *` and retrieve an element from it. Ideally, we should fully support the integer types, but I understand that's a lot more complicated than just supporting the native types.